sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,469 @@
+= sycl_ext_oneapi_group_load_store
+
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+:dpcpp: pass:[DPC++]
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+
+IMPORTANT: This specification is a draft.
+
+
+== Notice
+
+[%hardbreaks]
+Copyright (c) 2024 Intel Corporation.  All rights reserved.
+
+Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are trademarks
+of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc. used by
+permission by Khronos.
+
+== Dependencies
+
+This extension is written against the SYCL 2020 specification, Revision 8.
+All references below to the "core SYCL specification" or to section numbers in
+the SYCL specification refer to that revision.
+
+This extension also depends on the following other SYCL extensions:
+
+* link:../proposed/sycl_ext_oneapi_group_sort.asciidoc[
+  sycl_ext_oneapi_group_sort] (`group_with_scratchpad` class used as GroupHelper).
```

Comment:
I think this dependency only exists for the "Fixed-size array Overload" parts of the API.  Those parts of the extension also seem to have some unresolved issues in the specification.  Do we think these parts are important enough to do right away, or could they be moved to a "Future direction" section in the document?

It would be nice to implement the other parts of this spec soon, so that we can deprecate / remove the `sub_group::load` and `sub_group::store` functions from the DPC++ implementation.  I was wondering if we could implement these other parts sooner if we delay the "Fixed-size array Overload" parts until later.

Reply:
I think this is a good suggestion.

The single-value load/store functions are the most important for existing SYCL codes, and even just supporting these new functions would expose important new functionality for our GPUs (because the `sub_group::` versions don't allow selection of blocked/striped layout).

The fixed-size array overload parts are mainly useful to support patterns found in CUDA kernels that employ CUB.  They're important too, but could be implemented in a Phase 2.

Reply:
Agree, I moved everything related to group_with_scratchpad into Future direction section. @gmlueck, @Pennycook please take a look if it looks in place now